src/main.py

```diff
@@ -0,0 +1,47 @@
+import src.controllers as contr
+import src.pre_processing as pp
+import yfinance as yf
+import pandas as pd
+import numpy as np
+import os
+
+
+def correlation_testing():
+    tickers = pd.DataFrame(["GME", "AMC", "KOSS", "NAKD", "BBBY", "NOK", "VIX"], columns=['Ticker'])
+    sc = contr.StockController()
+    sc.init_stocks(tickers)
+    for stock_name, stock in sc.stocks.items():
+        stock.get_history()
+    corr = {stock_name: dict() for stock_name in sc.stocks.keys()}
+    norm_corr = {stock_name: dict() for stock_name in sc.stocks.keys()}
+    for stock_name, stock in sc.stocks.items():
+        for stock_name_2, stock_2 in sc.stocks.items():
+            if stock_name == stock_name_2:
+                continue
+            # length = min(len(stock.history['Close']), len(stock_2.history['Close']))
+            # length = 365
+            length = 150
+            stock_1_values = np.array(stock.history['Close'])[-length:]
+            stock_2_values = np.array(stock_2.history['Close'])[-length:]
+            norm_stock_1 = stock_1_values / np.sqrt(np.sum(stock_1_values**2))
+            norm_stock_2 = stock_2_values / np.sqrt(np.sum(stock_2_values**2))
+            corr[stock_name][stock_name_2] = float(np.corrcoef(stock_1_values, stock_2_values)[1, 0])
+            norm_corr[stock_name][stock_name_2] = float(np.corrcoef(norm_stock_1, norm_stock_2)[1, 0])
```

Comment:
Not sure here, but thinking about simplifying code:
- If `StockController` held e.g. a `DataFrame` with date as index, and a ticker per column, one should be able to use [DataFrame.corr()](https://pandas.pydata.org/pandas-docs/stable/reference/api/pandas.DataFrame.corr.html#pandas.DataFrame.corr)
- this can avoid explicit loops and ticker control
- it can also be automagically plotted with matplotlib into a heatmap correlation matrix

Reply:
I'm going to add my pennies here as well. 

- DataFrame.corr() is the easiest and more efficient way if you are only going for the coefficent 
- Maybe it makes sense to use scipy. It has built in correlation methods which also yield the p-value

Reply:
Thanks for the suggestions, reason I used numpy here is that DataFrame is Series and Series is np.array, so my logic was speed here. Let me find the [https://towardsdatascience.com/how-to-make-your-pandas-loop-71-803-times-faster-805030df4f06](link.)